tvm_vm/src/executor/token.rs

```diff
@@ -32,6 +51,84 @@ pub const KRBK: f64 = 0.675_f64;
 pub const KRBM: f64 = 0.1_f64;
 pub const MAX_FREE_FLOAT_FRAC: f64 = 1_f64 / 3_f64;
 
+pub const WASM_FUEL_MULTIPLIER: u64 = 8u64;
+struct MyState {
+    ctx: WasiCtx,
+    table: ResourceTable,
+}
+impl IoView for MyState {
+    fn table(&mut self) -> &mut ResourceTable {
+        &mut self.table
+    }
+}
+impl WasiView for MyState {
+    fn ctx(&mut self) -> &mut WasiCtx {
+        &mut self.ctx
+    }
+}
+
+// Async IO annotator for WASI. Do not use unless you know what you're doing.
+fn io_type_annotate<T: IoView, F>(val: F) -> F
+where
+    F: Fn(&mut T) -> IoImpl<&mut T>,
+{
+    val
+}
+// Sync annotator for WASI. Used in wasmtime linker
+fn type_annotate<T: WasiView, F>(val: F) -> F
+where
+    F: Fn(&mut T) -> WasiImpl<&mut T>,
+{
+    val
+}
+
+pub(super) fn split_to_chain_of_cells(input: Vec<u8>) -> Cell {
+    let cellsize = 120usize;
+    let len = input.len();
+    let mut cell_vec = Vec::<Vec<u8>>::new();
+    // Process the input in 1024-byte chunks
+    for i in (0..len).step_by(cellsize) {
+        let end = std::cmp::min(i + cellsize, len);
+        let chunk = &input[i..end];
+
+        // Convert slice to Vec<u8> and pass to omnom function
+        let chunk_vec = chunk.to_vec();
+        cell_vec.push(chunk_vec);
+
+        assert!(
+            cell_vec.last().expect("error in split_to_chain_of_cells function").len() == cellsize
+                || i + cellsize > len
+        );
+    }
+    let mut cell = BuilderData::with_raw(
```

Comment:
Should we panic?

Reply:
Rewrote the whole thing to return Result<Cell>. Must have had thsi code in tests before moving it here, since the unwrap.